src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDb = userService.findByEmail(email);
+        if (userFromDb.isEmpty()) {
+            throw new AuthenticationException("Can't authenticate user " + userFromDb);
+        }
+        User user = userFromDb.get();
+        String hashedPass = HashUtil.hashPassword(password, user.getSalt());
+        if (user.getPassword().equals(hashedPass)) {
+            return user;
+        }
+        throw new AuthenticationException("Can't authenticate user " + userFromDb);
```

Comment:
You need to simplify this method with only one if-block. You can saw this require into the checklist file.

Reply:
Agree